src/commands/analyze-revenue.cmd.ts

```diff
@@ -0,0 +1,178 @@
+    const fileConfig: AnalyzeRevenuesCommandOptions = {
+      ...JSON.parse(fs.readFileSync(`${options.inputsCacheDirPath}/config.json`).toString()),
+      inputsSource: InputsSource.FILES,
+      inputsCacheDirPath: options.inputsCacheDirPath,
+    }
+    
+    const config: AnalyzeRevenuesCommandOptions = { ...fileConfig, ...options }
+    this.logger.log(`Running "${COMMAND_NAME}" command...`, { ...config })
+
+    const snapshotValidatorsCollection = loadSnapshotValidatorsCollection(options.snapshotValidatorsFilePath)
+    const sourceDataOverrides = getValidatorOverrides(snapshotValidatorsCollection)
+
+    const dsSam = new DsSamSDK({ ...config })
+    const auctionDataCalculatedFromFixtures = await dsSam.run()
+    const auctionDataParsedFromFixtures: AuctionResult = JSON.parse(fs.readFileSync(options.resultsFixtureFilePath).toString())
+    console.log('Winning Total PMPE parsed from static results:', auctionDataParsedFromFixtures.winningTotalPmpe)
+    console.log('Winning Total PMPE calculated from static results:', auctionDataCalculatedFromFixtures.winningTotalPmpe)
+
+    const auctionValidatorsCalculatedWithOverrides = dsSam.transformValidators(await dsSam.getAggregatedData(sourceDataOverrides))
+    
+    return this.evaluateRevenueExpectationForAuctionValidators(auctionDataCalculatedFromFixtures.auctionData.validators, auctionValidatorsCalculatedWithOverrides)
+  }
+
+  evaluateRevenueExpectationForAuctionValidators = (validatorsBefore: AuctionValidator[], validatorsAfter: AuctionValidator[]): RevenueExpectation[] => {
+    const evaluation = []
+    for (const validatorBefore of validatorsBefore) {
+      const validatorAfter = validatorsAfter.find((v) => v.voteAccount === validatorBefore.voteAccount)
+      if (!validatorAfter) {
+        this.logger.warn("Validator not present in the snapshot!", { voteAccount: validatorBefore.voteAccount })
+        continue
+      }
+
+      const expectedNonBidPmpe = validatorBefore.revShare.inflationPmpe + validatorBefore.revShare.mevPmpe
+      const actualNonBidPmpe = validatorAfter.revShare.inflationPmpe + validatorAfter.revShare.mevPmpe
+
+      evaluation.push({
+        voteAccount: validatorBefore.voteAccount,
+        expectedInflationCommission: validatorBefore.inflationCommissionDec,
+        actualInflationCommission: validatorAfter.inflationCommissionDec,
+        expectedMevCommission: validatorBefore.mevCommissionDec,
+        actualMevCommission: validatorAfter.mevCommissionDec,
+        expectedNonBidPmpe,
+        actualNonBidPmpe,
+        expectedSamPmpe: expectedNonBidPmpe + (validatorBefore.revShare.auctionEffectiveBidPmpe ?? 0),
+        maxSamStake: validatorBefore.maxStakeWanted,
+        samStakeShare: validatorBefore.auctionStake.marinadeMndeTargetSol === 0 ? 1 : validatorBefore.auctionStake.marinadeSamTargetSol / (validatorBefore.auctionStake.marinadeMndeTargetSol + validatorBefore.auctionStake.marinadeSamTargetSol),
```

Comment:
Agreed on readability.

Unless I missed something, the expression should never be / 0, right?

The point of the expression is this:
1) Validator got 1000 SOL staked
2) Validator should get nothing out of MNDE
3) Validator should get nothing out of SAM
4) Validator's max bid is 0.1 (Still not enough to win the auction)
5) We still need to charge the bid in this case and pipelines need to be made aware of the fact that the bid is to be charged for all 1000 SOL

Reply:
cc @martinkrecek

Reply:
Ok, got it, keeping the check for MNDE stake is correct, rather than the other way around. 👍

And regarding the note on `0 / 0`, my comment doesn't make sense of course. 😄
I somehow iterated on the expression and as I was focusing on the division part (and switching between other parts of the PR), I somehow completely dropped the fact that the original expression also had a check for `0` at start (and it's obviously enough to have just one of the vars non-zero). 🤷‍♂️
Not sure how I managed to not catch it when posting the final comment, but resolving now. 🙃

Reply:
Good, we are on the same page, then.